.circleci/config.yml

```diff
@@ -136,391 +145,396 @@ jobs:
-
-      - store_artifacts:
-          path: ./docs/build/html
+            pip install pip --upgrade
+            pip install -r requirements.txt
+            pip install -r tests/requirements.txt
+            pip install dwave-preprocessing --no-index -f dist/ --no-deps --force-reinstall
+      - run: *unix-run-tests
 
-      - run:
-          name: doctest
-          command: |
-            . env/bin/activate
-            make -C docs/ doctest
+  test-sdist:
+    docker:
+      - image: circleci/python:3.9
 
+    steps:
+      - checkout
+      - attach_workspace:
+          at: dist
       - run:
-          name: linkcheck
+          name: install from sdist
           command: |
+            python -m venv env
             . env/bin/activate
-            make -C docs/ linkcheck
-
-  #
-  # cpp tests
-  #
+            pip install dist/dwave-preprocessing*.tar.gz
+      - run: *unix-run-tests
 
-  test-linux-cpp11:
-    docker:
-      # just use a python image, all we really want is debian
-      - image: circleci/python:3.8-buster
+  test-windows:
+    parameters:
+      python-version:
+        type: string
 
-    working_directory: ~/repo
+    executor:
+      name: win/default
 
     steps:
       - checkout
+      - attach_workspace:
+          at: dist
 
-      - restore_cache: &restore-cache-env
-          key: v2-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "tests/requirements.txt" }}-{{ checksum "docs/requirements.txt" }}-{{ .Environment.CIRCLE_JOB }}
-
-      - run: &create-virtualenv
-          name: Create virtual environment
+      - run:
+          name: install python and create venv
           command: |
-            python -m pip install virtualenv
-            python -m virtualenv env
+            nuget install python -Version << parameters.python-version >> -ExcludeVersion -OutputDirectory .
+            .\python\tools\python.exe --version
+            .\python\tools\python.exe -m venv env
 
-      - run: &install-requirements
-          name: Install requirements
+      - run:
+          name: install dependencies
           command: |
-            . env/bin/activate
-            pip install -r requirements.txt -r tests/requirements.txt -r docs/requirements.txt
-
-      - save_cache: &save-cache-env
-          key: v2-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "tests/requirements.txt" }}-{{ checksum "docs/requirements.txt" }}-{{ .Environment.CIRCLE_JOB }}
-          paths:
-            - env
+            env\Scripts\activate.ps1
+            python --version
+            pip install -r requirements.txt
+            pip install -r tests\requirements.txt
+            pip install dwave-preprocessing --no-index -f dist/ --force-reinstall --no-deps
 
       - run:
-          name: run cpp tests
+          name: run unittests
           command: |
-            . env/bin/activate
-            make -C testscpp/ --always-make
+            env\Scripts\activate.ps1
+            cd tests\
```

Comment:
Do we need to `cd` into `tests`?

Reply:
See https://github.com/dwavesystems/dwave-preprocessing/pull/38/files/28f3bbdeb54ec5ffa919670eb95fb41448874e41#r758667665